src/calendar/model/CalendarModel.ts

```diff
@@ -99,7 +102,8 @@ export class CalendarModel {
 		if (
 			existingEvent._ownerGroup !== groupRoot._id ||
 			newEvent.startTime.getTime() !== existingEvent.startTime.getTime() ||
-			!repeatRulesEqual(newEvent.repeatRule, existingEvent.repeatRule)
+			(!repeatRulesEqual(newEvent.repeatRule, existingEvent.repeatRule) &&
```

Comment:
the initial idea was that the check for the changed repeat rule might have failed because the exclusions changed, but that case is not grounds for recreating the event entirely, which is why we check that explicitly afterwards. 

but I generally agree that a changed repeat rule should not cause us to change the id, so we don't need to recreate.